### PR TITLE
feat: add transaction command-line interface

### PR DESCRIPTION
**Pull request by kogisin**
## Description

closes: #41 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Appropriate labels applied
- [x] Targeted PR against correct branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/farming/types/msgs.go [hunk 1]
```diff
@@ -56,13 +56,13 @@ func (msg MsgCreateFixedAmountPlan) ValidateBasic() error {
 		return sdkerrors.Wrapf(ErrInvalidPlanEndTime, "end time %s must be greater than start time %s", msg.EndTime.Format(time.RFC3339), msg.StartTime.Format(time.RFC3339))
 	}
 	if msg.StakingCoinWeights.Empty() {
-		return ErrEmptyStakingCoinWeights
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "staking coin weights must not be empty")
 	}
 	if err := msg.StakingCoinWeights.Validate(); err != nil {
```
Comment (dongsam) [inline x/farming/types/msgs.go:61]: Can we add validation for checking the sum of total weights == 1?

Comment (dongsam) [reply x/farming/types/msgs.go:61]: I think it should also be included in `MsgCreateRatioPlan`, `MsgCreateFixedAmountPlan`, `AddRequestProposals`, , `UpdateRequestProposals` and `func (plan BasePlan) Validate()`

**Review by dongsam — CHANGES_REQUESTED** [review]
LGTM in code level, I'll detail the actual test after #45 merged, 
It's trivial, but how about change the example value of `epoch_amount` to a more realistic value? amount 1 is not usual, and Can we add some detailed description for `staking_coin_weights` including It is decimal weight, not amount and the sum of total weight must be 1

**Review by hallazzang — CHANGES_REQUESTED** [review]

File: x/farming/types/msgs.go (hunk 1 above)
Comment (kogisin) [reply x/farming/types/msgs.go:61]: It looks like validation logic is not used for `Stake`, `Unstake`, `Harvest` as well. I'll add those  too.

File: x/farming/client/cli/tx.go [hunk 2]
```diff
@@ -174,9 +190,9 @@ func NewStakeCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "stake [amount]",
 		Args:  cobra.ExactArgs(1),
-		Short: "stake coins",
+		Short: "Stake coins into the farming plan",
```
[the remark this answers is not in the document]
Comment (kogisin) [reply x/farming/client/cli/tx.go:185]: Right, I'll update it to just `stake coins`

[the remark this answers is not in the document]
Comment (kogisin) [reply x/farming/client/cli/tx.go:185]: For long description, what do you think about adding this explanation? @hallazzang
```
Stake coins. You should check which plans are available on a network and stake coins accordingly.
```

[the remark this answers is not in the document]
Comment (hallazzang) [reply x/farming/client/cli/tx.go]: Looks good to me.

File: x/farming/keeper/msg_server.go [hunk 3]
```diff
@@ -29,6 +29,10 @@ var _ types.MsgServer = msgServer{}
 func (k msgServer) CreateFixedAmountPlan(goCtx context.Context, msg *types.MsgCreateFixedAmountPlan) (*types.MsgCreateFixedAmountPlanResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
```
Comment (hallazzang) [inline x/farming/keeper/msg_server.go:35]: As I know we don't need to call `ValidateBasic` since the SDK already called it under the hood.

**Review by dongsam — APPROVED** [review]

**Pull request merged**